File: angularapp/src/app/services/razorpay-payment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { API_URL } from '../constants/app.constant';
import { RazorpayPayment } from '../models/razorpay-payment';

@Injectable({
  providedIn: 'root'
})
export class RazorpayPaymentService {
  constructor(private http: HttpClient) {}

  createPayment(payment: RazorpayPayment): Observable<RazorpayPayment> {
    return this.http.post<RazorpayPayment>(`${API_URL}/payment/create`, payment);
  }
  createCashPayment(payment: RazorpayPayment): Observable<RazorpayPayment> {
    return this.http.post<RazorpayPayment>(`${API_URL}/payment/createCash`, payment);
  }

  updatePayment(id: number, payment: RazorpayPayment): Observable<RazorpayPayment> {
    return this.http.put<RazorpayPayment>(`${API_URL}/payment/${id}`, payment);
  }

  deletePayment(id: number): Observable<void> {
    return this.http.delete<void>(`${API_URL}/payment/${id}`);
  }

  getAllPayments(): Observable<RazorpayPayment[]> {
    return this.http.get<RazorpayPayment[]>(`${API_URL}/payment`);
  }

  getPaymentById(id: number): Observable<RazorpayPayment> {
    return this.http.get<RazorpayPayment>(`${API_URL}/payment/${id}`);
  }

  
 getPaymentByAppointmentId(appointmentId: number): Observable<RazorpayPayment> {
  return this.http.get<RazorpayPayment>(`${API_URL}/payment/appointment/${appointmentId}`);
}

}

File: springapp/src/main/java/com/examly/springapp/service/RazorpayPaymentService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.dto.RazorpayPaymentDto;

public interface RazorpayPaymentService {


    RazorpayPaymentDto createPayment(RazorpayPaymentDto dto);

    RazorpayPaymentDto getPaymentById(Long id);

    List<RazorpayPaymentDto> getAllPayments();

    RazorpayPaymentDto updatePayment(Long id, RazorpayPaymentDto dto);

    void deletePayment(Long id);

    RazorpayPaymentDto getPaymentByAppointmentId(Long appointmentId);
     RazorpayPaymentDto createCashPayment(RazorpayPaymentDto dto);


}
